apps/frontend/src/components/generic/CopyButton.vue

```diff
@@ -26,12 +17,10 @@ export default class CopyButton extends Vue {
   @Prop({required: false, default: 'mdi-clipboard-outline'})
   readonly icon!: string;
 
-  onCopy() {
-    SnackbarModule.notify('Text copied to your clipboard');
-  }
-
-  onCopyFailure() {
-    SnackbarModule.failure('Failed to copy to your clipboard');
+  copy() {
+    this.$copyText(this.text, document.querySelector('.v-dialog') || undefined)
+      .then(() => SnackbarModule.notify('Text copied to your clipboard'))
+      .catch(() => SnackbarModule.failure('Failed to copy to your clipboard'));
```

Comment:
Was this implementation of Vue added on a later version - not sure why it wasn't implemented this way from the start!